### PR TITLE
[ET-372]: WIP Rename delivery package to automate - **Do not merge**

### PR DESCRIPTION
**Pull request by PrajaktaPurohit**
**Do not merge**

### CODE REVIEW

**Comment by schisamo** [comment]
@wrightp When you are back tomorrow please give @PrajaktaPurohit some feedback on the direction she is taking on these changes?

File: lib/mixlib/install/product.rb [hunk 1]
```diff
@@ -255,10 +279,34 @@ def lookup(key, version = :latest)
...
+    install_path do |v|
+      if v < version_for("9.0.0")
+      #TODO: Once we change the paths of the config files for automate,
+      #      update the following path to /opt/automate
+        "/opt/delivery"
+      else
+        "/opt/delivery"
+      end
+    end
   end
```
Comment (schisamo) [inline lib/mixlib/install/product.rb]: We can probably completely kill the `delivery` product entry now. Keeping it around will probably just cause confusion. If a consumer of mixlib-install via something like chef-ingredient uses the old naming they can just update the their use of the `delivery` product key to `automate`.

File: lib/mixlib/install/product.rb [hunk 2]
```diff
@@ -193,9 +193,33 @@ def lookup(key, version = :latest)
 
   product "automate" do
     product_name "Chef Automate"
-    package_name "delivery"
-    ctl_command "delivery-ctl"
-    config_file "/etc/delivery/delivery.rb"
+    package_name do |v|
+    #TODO: Change the version to the promoted release version of automate package
+      v < version_for("8.0.0") ? "delivery" : "automate"
```
Comment (PrajaktaPurohit) [inline lib/mixlib/install/product.rb]: these versions will be determined once we have an actual release of automate. (the assumption here is that, thatversion will be greator that any of the delivery versions relesed so far)
for now: 8.0.0 is the first version of automate to be released
4.4.4 is the version of delivery where automate-ctl was enabled

**Comment by schisamo** [comment]
Related Omnitruck PR: https://github.com/chef/omnitruck/pull/203

File: PRODUCT_MATRIX.md [hunk 3]
```diff
@@ -10,7 +10,7 @@
 | Chef Server HA Provisioning for AWS | chef-server-ha-provisioning |
 | Chef Development Kit | chefdk |
 | Chef Compliance | compliance |
-| Delivery | delivery |
+| Chef Automate | delivery |
```
Comment (wrightp) [inline PRODUCT_MATRIX.md]: We should take a hard stance on this. The naming of `delivery` should just be known as `automate` from here on out. Update the key to be `automate.`

File: lib/mixlib/install/product.rb [hunk 4]
```diff
@@ -193,9 +193,33 @@ def lookup(key, version = :latest)
 
   product "automate" do
```
Comment (wrightp) [inline lib/mixlib/install/product.rb]: Treat the `automate` and `delivery` products as separate entities. No backwards compat logic for which product name they specify. That will remove the copy/paste issue here and logic that is needed here like the ctl name change will be more clear.

**Comment by wrightp** [comment]
Closing PR in favor of #177

**Pull request closed without merging**